x-pack/plugins/fleet/public/applications/fleet/sections/data_stream/list_page/components/data_stream_row_actions.tsx

```diff
@@ -33,52 +35,74 @@ export const DataStreamRowActions = memo<{ datastream: DataStream }>(({ datastre
     defaultMessage: 'View dashboards',
   });
 
-  if (!dashboards || dashboards.length === 0) {
+  const viewServiceInApmActionTitle = i18n.translate(
+    'xpack.fleet.dataStreamList.viewServiceInApmActionText',
+    {
+      defaultMessage: 'View Service in APM',
+    }
+  );
+
+  const { isReady, href } = useAPMServiceHref(datastream);
+
+  if (isReady && href) {
```

Comment:
IMHO there's a little too much branching here. Can we simplify?

Reply:
Do we only link to the APM app if there's one service in a data stream? That will be the most common scenario for now. Maybe we should link to the service inventory if there's multiple (or none)?

Reply:
For now, yes, we only link to the APM app if there is exactly 1 service attached to a given data stream, I'll follow up with Alex as I feel this is more of a product decision?

Reply:
Yes, for sure. But let's talk about it before merging, because the default is that everything ends up in one data stream so this feature has little value if we're not linking to the APM app at all if there is data for multiple services in a data stream.

Reply:
I changed it to link to the service inventory if a datastream has zero or more than one services https://github.com/elastic/kibana/pull/130251/commits/364d53f892c45cc233ff97ff14f2abe78e63716e